source/particles/particle_handler.cc

```diff
@@ -547,15 +547,11 @@ namespace Particles
     const std::set<types::subdomain_id> ghost_owners =
       triangulation->ghost_owners();
 
-    for (auto ghost_domain_id = ghost_owners.begin();
-         ghost_domain_id != ghost_owners.end();
-         ++ghost_domain_id)
-      moved_particles[*ghost_domain_id].reserve(
+    for (const unsigned int ghost_owner : ghost_owners)
```

Comment:
This type isn't right: it should be `types::subdomain_id`. I guess we should use auto here.

Reply:
Yes, this is difficult to get right with the `clang-tidy` fix...